src/elysium/ecdsa_signature.cpp

```diff
@@ -17,32 +17,39 @@ ECDSASignature::ECDSASignature(secp256k1_ecdsa_signature const &sig)
 {
 }
 
-ECDSASignature::ECDSASignature(ECDSAContext const &context, unsigned char const *signature, size_t len)
-    : valid(false)
+ECDSASignature ECDSASignature::Parse(ECDSAContext const &context, unsigned char const *signature, size_t len)
 {
+    ECDSASignature sig;
```

Comment:
Why not use `secp256k1_ecdsa_signature` instead? With this way we don't need to update `valid` flag by ourself.